autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/FreeIpaCommunicator.java

```diff
@@ -33,17 +42,36 @@ public FreeIpaCommunicator(UserV1Endpoint userV1Endpoint) {
     public SyncOperationStatus synchronizeAllUsers(SynchronizeAllUsersRequest request) {
         LOGGER.info("Invoking freeIpa user sync request: {}", request);
         String envCrn = request.getEnvironments().iterator().next();
-        SyncOperationStatus lastSyncStatus = ThreadBasedUserCrnProvider.doAsInternalActor(
-                internalCrnGeneratorFactory.autoscale().getInternalCrnForServiceAsString(),
-                () -> userV1Endpoint.getLastSyncOperationStatus(envCrn));
-        if (SynchronizationStatus.RUNNING.equals(lastSyncStatus.getStatus())) {
-            LOGGER.info("There is a user sync operation already running for environment: {} with operationId: {}, " +
-                    "skipping request to trigger another user sync", envCrn, lastSyncStatus.getOperationId());
-            return lastSyncStatus;
-        } else {
-            return ThreadBasedUserCrnProvider.doAsInternalActor(
+        try {
+            SyncOperationStatus lastSyncStatus = ThreadBasedUserCrnProvider.doAsInternalActor(
+                    internalCrnGeneratorFactory.autoscale().getInternalCrnForServiceAsString(),
+                    () -> userV1Endpoint.getLastSyncOperationStatus(envCrn));
+            if (SynchronizationStatus.RUNNING.equals(lastSyncStatus.getStatus())) {
+                LOGGER.info("There is a user sync operation already running for environment: {} with operationId: {}, " +
+                        "skipping request to trigger another user sync", envCrn, lastSyncStatus.getOperationId());
+                return lastSyncStatus;
+            } else {
+                return invokeFreeipaUserSyncAndHandleException(request, envCrn);
+            }
+        } catch (Exception ex) {
```

Comment:
Why is this catch block required here?
As things are set up right now - if there's an error from userV1Endpoint.synchronizeAllUsers - it is going to end up being wrapped in 2 RuntimExceptions.

Reply:
Changed this